universum/modules/structure_handler.py

```diff
@@ -140,7 +140,12 @@ def execute_one_step(self, configuration: Step,
 
         process.start()
         if not configuration.background:
-            process.finalize()
+            try:
+                process.finalize()
+            except StepException:
+                if configuration.if_failed:  # conditional step with next step configured at `if_failed` event
```

Comment:
This is a generic function for executing single steps.
Could you please consider moving this conditional-steps-specific functionality to the respective functions?